boot.go

```diff
@@ -11,11 +11,15 @@ import (
 	"github.com/deis/builder/pkg/cleaner"
 	"github.com/deis/builder/pkg/conf"
 	"github.com/deis/builder/pkg/gitreceive"
-	"github.com/deis/builder/pkg/gitreceive/storage"
 	"github.com/deis/builder/pkg/healthsrv"
 	"github.com/deis/builder/pkg/sshd"
 	"github.com/deis/builder/pkg/sys"
 	pkglog "github.com/deis/pkg/log"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	_ "github.com/docker/distribution/registry/storage/driver/azure"
+	"github.com/docker/distribution/registry/storage/driver/factory"
+	_ "github.com/docker/distribution/registry/storage/driver/gcs"
+	_ "github.com/docker/distribution/registry/storage/driver/s3-aws"
```

Comment:
A solid place to start with back registration is [here](https://github.com/docker/distribution/blob/master/registry/storage/driver/s3-aws/s3.go#L88). The factory is registered such that the driver can be instantiated by name and opaque configuration values. The anonymous imports (ie `_`) force the registration via the `init` method and make it available. It allows one to have [different mains](https://github.com/docker/distribution/blob/master/cmd/registry/main.go) to compile in different sets of driver plugins. We used to use this, along with build tags, to control which drivers were compiled into the binary. This is a pattern that is leveraged in several places, such as `crypto` hash functions and `database/sql`.

What I meant by my word soup was that we can separate the back registration from the driver implementations. Registry specific configuration is present in a lot of the driver packages that we might want to have a closer look at. The exact split is unclear, but we would go with what would make it broadly useful.